pkg/storage/unified/resource/datastore.go

```diff
@@ -274,20 +420,221 @@ func ParseKey(key string) (DataKey, error) {
+		Namespace: namespace, // Empty string if not specified, which will list all namespaces
+	}
+
+	// Maps to track counts per namespace for this group/resource
+	namespaceCounts := make(map[string]map[string]bool) // namespace -> resource name -> exists (not deleted)
+	namespaceVersions := make(map[string]int64)         // namespace -> latest resource version
+
+	// Track current resource being processed
+	var currentResourceKey string
+	var lastDataKey *DataKey
+
+	// Helper function to process the last seen resource
+	processLastResource := func() {
+		if lastDataKey != nil {
+			// Initialize maps if needed
+			if _, exists := namespaceCounts[lastDataKey.Namespace]; !exists {
+				namespaceCounts[lastDataKey.Namespace] = make(map[string]bool)
+				namespaceVersions[lastDataKey.Namespace] = 0
+			}
+
+			// Track whether this resource name exists (not deleted)
+			namespaceCounts[lastDataKey.Namespace][lastDataKey.Name] = lastDataKey.Action != DataActionDeleted
+
+			// Update to latest resource version seen
+			if lastDataKey.ResourceVersion > namespaceVersions[lastDataKey.Namespace] {
+				namespaceVersions[lastDataKey.Namespace] = lastDataKey.ResourceVersion
+			}
+		}
+	}
+
+	// List all keys using the existing Keys method
+	for dataKey, err := range d.Keys(ctx, listKey) {
+		if err != nil {
+			return nil, err
+		}
+
+		// Create unique resource identifier (namespace/group/resource/name)
+		resourceKey := fmt.Sprintf("%s/%s/%s/%s", dataKey.Namespace, dataKey.Group, dataKey.Resource, dataKey.Name)
+
+		// If we've moved to a different resource, process the previous one
+		if currentResourceKey != "" && resourceKey != currentResourceKey {
+			processLastResource()
+		}
+
+		// Update tracking variables for the current resource
+		currentResourceKey = resourceKey
+		lastDataKey = &dataKey
+	}
+
+	// Process the final resource
+	processLastResource()
+
+	// Convert namespace counts to ResourceStats
+	stats := make([]ResourceStats, 0, len(namespaceCounts))
+	for ns, names := range namespaceCounts {
+		// Count how many names actually exist (not deleted)
+		count := int64(0)
+		for _, exists := range names {
+			if exists {
+				count++
+			}
+		}
+
+		// Skip if count is below or equal to minimum
+		if count <= int64(minCount) {
+			continue
+		}
+
+		stats = append(stats, ResourceStats{
+			NamespacedResource: NamespacedResource{
+				Namespace: ns,
+				Group:     groupResource.Group,
+				Resource:  groupResource.Resource,
+			},
+			Count:           count,
+			ResourceVersion: namespaceVersions[ns],
+		})
+	}
+
+	return stats, nil
+}
+
+// getGroupResources returns all unique group/resource combinations in the data store.
+// It efficiently discovers these by using the key ordering and PrefixRangeEnd to jump
+// between different group/resource prefixes without iterating through all keys.
+// Results are cached to improve performance.
+func (d *dataStore) getGroupResources(ctx context.Context) ([]GroupResource, error) {
```

Comment:
For the future, could we get these resources from the k8s apiserver... somehow? For example, locally, if I do `kubectl --kubeconfig=./grafana.kubeconfig api-resources` I can see all registered resources.